handlers.go

```diff
@@ -239,32 +240,49 @@ func VerifyReset(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprintf(w, "%s", buildResetCompletePage(result.Roll, result.Email))
 }
 
-func GetEmail(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+func GetHandlerFunc(input, output string) func (w http.ResponseWriter, r *http.Request) {
 
-	if !PublicApiAuthenticate(r.Header.Get("Authorization")) {
-		http.Error(w, ERROR_UNAUTH, 401)
-		return
-	}
+	return func (w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json; charset=utf-8")
 
-	var roll string
-	if r.Method == "POST" {
-		r.ParseForm()
-		roll = r.PostForm.Get("roll")
-	}
+		if !PublicApiAuthenticate(r.Header.Get("Authorization")) {
```

Comment:
I think we can abstract the checks into independent components here.
Moreover, as the name of the function suggests, it is a `HandlerFunction` rather than being a direct `Handler` itself. Thus, it's job should be to handle the closure values and call appropriate Handlers.

```
func GetHandlerFunc(input, output string) func (w http.ResponseWriter, r *http.Request) {
    return func (w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Content-Type", "application/json; charset=utf-8")
        err := checkAuthentication(r)
        if err != nil {
            // raise error
        }
        switch(output):
        case "roll": 
               ret_value := HandleRoll(w, r)
        case "email":
               ret_value := HandleEmail(w, r)

        fmt.Fprintf(w, ret_value)
    }
}
```

Reply:
that looks way better. I have to get into the way of doing go things more idiomatically.

Reply:
Reference: https://github.com/campoy/go-web-workshop/blob/master/section03/README.md#avoiding-repetition

Reply:
Thanks for the reference. I have some clean-up work to do on the code here. (First, make this a library)

I am merging this in for now. Meanwhile, I will work on the changes.

Reply:
Makes sense.

Meanwhile, I was thinking of making some fixes to the code.
Since you are refactoring the code now, would it be advisable if I proceed with it?